### PR TITLE
Closes #112 | Implement Dataloader for Wisesight Thai Corpus

### LINKED ISSUE #112: Create dataset loader for Wisesight Thai Sentiment Corpus
Dataloader name: `wisesight_thai_sentiment/wisesight_thai_sentiment.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?wisesight_thai_sentiment

|     Dataset| wisesight_thai_sentiment |
|-------------|---|
| Description | This dataset comprises 26,737 Thai social media texts from 2016 to early 2019, primarily informal and conversational, including some news headlines and advertisements. It is split across positive, negative, and neutral sentiments. The content mainly relates to consumer products and services, with a portion on current affairs. It includes only publicly available internet messages, specifically public comments on Facebook pages. Private communications and group messages are excluded. The corpus doesn't statistically represent any language register, contains modifications like personal data removal, and maintains original (mis)spellings and emojis. Messages over 2,000 characters, non-Thai, or duplicates are omitted. |
| Subsets | - |
| Languages | tha |
| Tasks | Sentiment Analysis, Aspect Based Sentiment Analysis |
| License | Creative Commons Zero v1.0 Universal (cc0-1.0) |
| Homepage | https://github.com/PyThaiNLP/wisesight-sentiment |
| HF URL | - |
| Paper URL | https://github.com/PyThaiNLP/wisesight-sentiment |

### PR DESCRIPTION
**Pull request by bp-high**
Closes #112 

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [ ] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [ ] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+_SOURCE_VIEW_NAME = DEFAULT_SOURCE_VIEW_NAME
+_UNIFIED_VIEW_NAME = DEFAULT_SEACROWD_VIEW_NAME
+
+_HOMEPAGE = "https://github.com/PyThaiNLP/wisesight-sentiment"
+
+_LANGUAGES = ["tha"]  # We follow ISO639-3 language code (https://iso639-3.sil.org/code_tables/639/data)
+
+
+_LICENSE = "Creative Commons Zero v1.0 Universal license"
```
Comment (ryanignatius) [inline seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py:80]: please use `Licenses` enum from seacrowd.utils.constants

File: seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+                    "category": datasets.features.ClassLabel(names=["pos", "neu", "neg", "q"]),
+                }
+            )
+        elif self.config.schema == "seacrowd_text":
+            features = schemas.text_features(["pos", "neu", "neg", "q"])
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage="https://github.com/PyThaiNLP/wisesight-sentiment",
```
Comment (ryanignatius) [inline seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py:142]:
```suggestion
            homepage=_HOMEPAGE,
```

File: seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+            citation=_CITATION,
```
Comment (ryanignatius) [inline seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py:143]: also add the license here
```suggestion
            license=_LICENSE,
            citation=_CITATION,
```

**Review by ryanignatius — CHANGES_REQUESTED** [review]

File: seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py (hunk 1 above)
Comment (bp-high) [reply seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py:80]: Okay

File: seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py (hunk 2 above)
Comment (bp-high) [reply seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py:142]: Okay

File: seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py (hunk 3 above)
Comment (bp-high) [reply seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py:143]: Okay

File: seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+#  provided by the original dataset as a version goes.
+_SOURCE_VERSION = "1.0.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+
+class WisesightSentiment(datasets.GeneratorBasedBuilder):
+    """Wisesight Sentiment Corpus: Social media messages in Thai language with sentiment category (positive, neutral, negative, question)"""
+
+    _DOWNLOAD_URL = "https://github.com/PyThaiNLP/wisesight-sentiment/raw/master/huggingface/data.zip"
```
[the remark this answers is not in the document]
Comment (bp-high) [reply seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py:104]: Okay

**Comment by bp-high** [comment]
@ryanignatius have made the respective changes you suggested in your review. Thanks for the suggestions.

File: seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+#  provided by the original dataset as a version goes.
+_SOURCE_VERSION = "1.0.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+
+class WisesightSentimentDataset(datasets.GeneratorBasedBuilder):
+    """Wisesight Sentiment Corpus: Social media messages in Thai language with sentiment category (positive, neutral, negative, question)"""
+
+    _DOWNLOAD_URL = _URLS["_DATASETNAME"]
```
Comment (ryanignatius) [inline seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py:104]: the `_DATASETNAME` should be using variable and not the constant
```suggestion
    _DOWNLOAD_URL = _URLS[_DATASETNAME]
```

File: seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from pathlib import Path
+from typing import Dict, List, Tuple
```
Comment (ryanignatius) [inline seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py:18]: please remove unused import

File: seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+
+
+_LOCAL = False
+
+
+_URLS = {
+    _DATASETNAME: "https://github.com/PyThaiNLP/wisesight-sentiment/raw/master/huggingface/data.zip",
+}
+
+# TODO: add supported task by dataset. One dataset may support multiple tasks
```
Comment (ryanignatius) [inline seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py:90]: please remove this TODO comment

File: seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+_SUPPORTED_TASKS = [Tasks.SENTIMENT_ANALYSIS]  # example: [Tasks.TRANSLATION, Tasks.NAMED_ENTITY_RECOGNITION, Tasks.RELATION_EXTRACTION]
+
+# TODO: set this to a version that is associated with the dataset. if none exists use "1.0.0"
+#  This version doesn't have to be consistent with semantic versioning. Anything that is
+#  provided by the original dataset as a version goes.
```
Comment (ryanignatius) [inline seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py:95]: please remove this TODO comment

**Review by ryanignatius — COMMENTED** [review]
I have reviewed and tested again,
only need some additional minor adjustments

File: seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py (hunk 5 above)
Comment (bp-high) [reply seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py:104]: Oh yep sorry

File: seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py (hunk 6 above)
Comment (bp-high) [reply seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py:18]: okay

File: seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py (hunk 8 above)
Comment (bp-high) [reply seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py:95]: done

File: seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py (hunk 7 above)
Comment (bp-high) [reply seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py:90]: done

**Review by ryanignatius — APPROVED** [review]
Reviewed and tested, LGTM
Let's wait for other reviewer

**Review by MJonibek — CHANGES_REQUESTED** [review]
Everything works. 

The minor changes are needed:
1. Please run command to format file
 
make check_file=seacrowd/sea_datasets/wisesight_thai_sentiment/wisesight_thai_sentiment.py 

2. And can you add init file please.

**Comment by bp-high** [comment]
@MJonibek Have implemented the steps you suggested.

**Review by MJonibek — APPROVED** [review]
Checked, LGTM

**Pull request merged**